core/services/workflows/syncer/workflow_registry.go

```diff
@@ -195,42 +226,35 @@ func NewWorkflowRegistry(
 	for _, opt := range opts {
 		opt(wr)
 	}
-	return wr
+
+	return wr, nil
 }
 
 // Start begins the workflowRegistry service.
 func (w *workflowRegistry) Start(_ context.Context) error {
 	return w.StartOnce(w.Name(), func() error {
-		ctx, cancel := w.stopCh.NewCtx()
+		ctx, _ := w.stopCh.NewCtx()
```

Comment:
We shouldn't leave the cancelFn dangling; it spawns a goroutine that we're leaking here. You need to encapsulate the code in a way that allows you to call it (eg. by spawning a goroutine that blocks)